### ISSUE TITLE
How to serve statics ?

### ISSUE THREAD

**Issue by valentin2105** [tootsuite/mastodon#767]
Hi,

I tried to deploy Mastodon using Docker-compose, exposed via a Nginx container who act as a reverse-proxy.  ( I mount mastodon/public as a volume in this container).

I use the same nginx.conf from the Production Wiki.

When I load my instance (in HTTPS with Let's encrypt), I got problems with statics (css/js) who give me few 404 errors.

My folder ../mastodon/public/assets is empty, is that normal ? 

Thanks for your help.

**Comment by rhaamo** [comment]
Statics should be built using ```RAILS_ENV=production bundle exec rails assets:precompile``` this should also be executed any time you updates mastodon and there is assets changes (js, images.), but I don't have used Docker so I can't say if it's executed properly or not for this specific point.

I have the same kind of setup (except docker) and I just rsync the whole ```public``` directory to one in the proxy server each update and use ```root /srv/http/whatever.tld/public;``` in the vhost.

**Comment by Gargron** [comment]
Above comment is correct. Let us know if that solves it!

**Comment by valentin2105** [comment]
@Gargron @rhaamo Thanks for your answer.

After generate assets, the folder ./mastodon/public/assets is provided with assets  (like application-3543534634....css) but when I load the homepage with this directive in my Nginx.conf ( ` root /srv/mastodon/public;`)  I got few 404's too (no CSS). 

For example :

`...xyz/images/logo.png 404 (Not Found)`

And in the public folder, I got : 

`assets/logo-5701d8d94bca66668f49a2803c984e960dfe0bf93a71f2d86be9455e66505cee.png`

I don't understand where I go wrong. 
Thanks for helping.
